framework/decode/vulkan_object_info.h

```diff
@@ -268,6 +268,8 @@ struct DeviceInfo : public VulkanObjectInfo<VkDevice>
     graphics::VulkanDevicePropertyFeatureInfo property_feature_info;
 
     std::unordered_map<uint32_t, VkDeviceQueueCreateFlags> queue_family_creation_flags;
+    std::unordered_map<VkQueue, uint32_t>                  queue_family_indices;
+    std::unordered_map<VkQueue, uint32_t>                  queue_indices;
```

Comment:
I think putting `queue_family_index` and `queue_index` in `QueueInfo` is better.

Reply:
The reason I didn't add it to the queue is this is defined during vkGetDeviceQueue and the logic to create the QueueInfo occurs after the override function is called.  So the only Info structure I had access to is the DeviceInfo struct.  Is there a way to have a `PostOverride` function?

```

void VulkanReplayConsumer::Process_vkGetDeviceQueue(
    const ApiCallInfo&                          call_info,
    format::HandleId                            device,
    uint32_t                                    queueFamilyIndex,
    uint32_t                                    queueIndex,
    HandlePointerDecoder<VkQueue>*              pQueue)
{
    auto in_device = GetObjectInfoTable().GetDeviceInfo(device);
    if (!pQueue->IsNull()) { pQueue->SetHandleLength(1); }
    QueueInfo handle_info;
    pQueue->SetConsumerData(0, &handle_info);

    OverrideGetDeviceQueue(GetDeviceTable(in_device->handle)->GetDeviceQueue, in_device, queueFamilyIndex, queueIndex, pQueue);

    AddHandle<QueueInfo>(device, pQueue->GetPointer(), pQueue->GetHandlePointer(), std::move(handle_info), &VulkanObjectInfoTable::AddQueueInfo);
}
```

Reply:
You can use 
`auto queue_info = reinterpret_cast<QueueInfo*>(pQueue->GetConsumerData(0));` 
to get `queue_info`  in `OverrideGetDeviceQueue`. 

You  can see
```
QueueInfo handle_info;
pQueue->SetConsumerData(0, &handle_info);
...
AddHandle<QueueInfo>(device, pQueue->GetPointer(), pQueue->GetHandlePointer(), std::move(handle_info), &VulkanObjectInfoTable::AddQueueInfo);
```
in `Process_vkGetDeviceQueue`.

`handle_info` is moved into table in  AddHandle.

Reply:
That's what I was hoping for!  Thanks @locke-lunarg

Reply:
Done.